backend/src/game/dice_handler.py

```diff
@@ -0,0 +1,24 @@
+import backend.src.game.dice as dice
+import backend.src.game.constants as constants
+
+
+class DiceHandler(object):
+    dice_values = list()
+    __re_rolls_left = constants.DEFAULT_RE_ROLL_COUNT
+
+    def start_turn(self):
+        self.dice_values = dice.roll_many(constants.DEFAULT_DICE_COUNT)
+        self.__re_rolls_left = constants.DEFAULT_RE_ROLL_COUNT
+
+    def re_roll_dice(self, indexes_of_dice_to_re_roll):
+        if self.__re_rolls_left > 0:
```

Comment:
If you're not going to raise an exception for this and go on, instead of nesting everything, you could just return out at this point.

Reply:
100% so much yes